src/Features/CSharp/Portable/ConvertProgram/ConvertProgramTransform_ProgramMain.cs

```diff
@@ -84,11 +98,29 @@ public static async Task<Document> ConvertToProgramMainAsync(Document document,
             if (method.ParameterList.Parameters.Count == 1 && method.ParameterList.Parameters[0].Type is ArrayTypeSyntax arrayType)
                 method = method.ReplaceNode(arrayType.ElementType, PredefinedType(Token(SyntaxKind.StringKeyword)));
 
-            return FixupComments(generator.ClassDeclaration(
-                WellKnownMemberNames.TopLevelStatementsEntryPointTypeName,
-                accessibility: useDeclaredAccessibity ? programType.DeclaredAccessibility : Accessibility.NotApplicable,
-                modifiers: hasExistingPart ? DeclarationModifiers.Partial : DeclarationModifiers.None,
-                members: new[] { method }).WithLeadingTrivia(leadingTrivia));
+            ClassDeclarationSyntax newClassDeclaration = null!;
```

Comment:
```suggestion
            ClassDeclarationSyntax? newClassDeclaration = null;
```

Reply:
`newClassDeclaration` will be enforced to not be null later. If `oldClassDeclaration` is null, then we generate `newClassDeclaration` entirely, otherwise it will be taken as a modified `oldClassDeclaration` with new `Main` method member added. So this is kinda workaround to not bother with `newClassDeclaration` potentially being null when returning it.

Reply:
@DoctorKrolic The compiler should be aware that `newClassDeclaration` is assigned to non-null and shouldn't produce warnings. Do you see any warnings if you use `ClassDeclarationSyntax? newClassDeclaration = null;`?

Reply:
Verified it, compiler is really smart here, I am impressed) I thought, it would produce a warning